mne/decoding/transformer.py

```diff
@@ -520,3 +520,89 @@ def transform(self, epochs_data, y=None):
                                      picks=self.picks, n_jobs=self.n_jobs,
                                      copy=False, verbose=False)
         return epochs_data
+
+
+class UnsupervisedSpatialFilter(TransformerMixin):
+    """Fit and transform with an unsupervised spatial filtering across time
+    and samples.
+
+    Parameters
+    ----------
+    estimator : scikit-learn estimator
+        Estimator using some decomposition algorithm.
+    average : bool, default False
+        If true, average of X is used in estimator's fit.
+    """
+    def __init__(self, estimator, average=False):
+        for attr in ('fit', 'transform', 'fit_transform'):
```

Comment:
It s a method that allows using sklearn clone. This is useful for parallelization for example